feat: Add three command bar actions for switching the Light/Dark Mode

nrysk opened the pull request:
This PR adds three new actions to the Command Bar, allowing users to switch the browser's theme directory. This is useful for web developers checking appearance in different themes.

The following commands are added:
- `Switch to Automatic Appearance`
- `Switch to Light Mode`
- `Switch to Dark Mode`

These commands are dynamically hidden if they are already active.

**Note of Naming**: command naming was inspired by Arc Browser

## Screenshot
**Switch to Automatic Appearance**
<img width="1357" height="832" alt="image" src="https://github.com/user-attachments/assets/0bda53a7-4c26-4478-802c-ac7f17c9d16b" />

**Switch to Light Mode**
<img width="1313" height="788" alt="image" src="https://github.com/user-attachments/assets/d14d2cd2-5751-49eb-88eb-41f58e0249c8" />

**Switch to Dark Mode**
<img width="1357" height="832" alt="image" src="https://github.com/user-attachments/assets/ff16c109-0db9-4de6-a4e6-f4fc602cf7d1" />

nrysk committed:
feat: Add three command bar actions for switching the Light/Dark Mode

mr-cheffy commented on src/zen/common/zen-sets.js:
```diff
@@ -129,6 +129,21 @@ document.addEventListener(
+          }
+          case 'cmd_zenSwitchDarkMode': {
+            Services.prefs.setIntPref('zen.view.window.scheme', 0);
+            Services.prefs.setIntPref('layout.css.prefers-color-scheme.content-override', 0);
```
No need to change this if `zen.view.window.scheme` is already being changed

nrysk replied:
I've removed the unnecessary `layout.css.prefers-color-scheme.content-override` call.

mr-cheffy commented on src/zen/urlbar/ZenUBGlobalActions.sys.mjs:
```diff
@@ -153,6 +153,30 @@ const globalActionsTemplate = [
+    command: 'cmd_zenSwitchAutomaticAppearance',
+    icon: 'chrome://browser/skin/zen-icons/sparkles.svg',
+    isAvailable: () => {
+      return Services.prefs.getIntPref('zen.view.window.scheme') !== 2;
```
Would be nice to have this as a lazy pref for better performance

nrysk replied:
I've implemented the `ThemeCache` for the lazy pref

nrysk committed:
refactor(perf): Address review feedback for theme command

Remove unnecessary `SetIntPref` and add `ThemeCache` to read lazy preference

nrysk commented:
@mr-cheffy
I've applied your feedback!

* Removed the unnecessary `layout.css.prefers-color-scheme.content-override` call.
* Implemented the `ThemeCache` for the lazy pref pattern directly within `ZenUBGlobalActions.sys.mjs`.

I wasn't 100% sure if this was the most appropriate file for this logic. If there's a better place for it, please let me know and I'd be happy to move it.

mr-cheffy commented on src/browser/base/content/zen-commands.inc.xhtml:
```diff
@@ -61,4 +61,8 @@
   <command id="cmd_zenTogglePinTab" />
   <command id="cmd_zenCloseUnpinnedTabs" />
+
+  <command id="cmd_zenSwitchAutomaticAppearance" />
```
Now that I think about it, we don't need to make so much boilerplate for this. We shoudn't be using commands, we should rather use inline commands.

Like this: 
https://github.com/zen-browser/desktop/blob/96c6ec658cff4ec8b69192375ce8fbfc68737a29/src/zen/urlbar/ZenUBGlobalActions.sys.mjs#L37

mr-cheffy commented on src/zen/common/zen-sets.js:
```diff
@@ -129,6 +129,18 @@ document.addEventListener(
             gZenWorkspaces.unloadWorkspace();
             break;
           }
+          case 'cmd_zenSwitchAutomaticAppearance': {
```
As said in the comments above, we don't really need this

mr-cheffy commented on src/zen/urlbar/ZenUBGlobalActions.sys.mjs:
```diff
@@ -6,6 +6,26 @@ function isNotEmptyTab(window) {
   return !window.gBrowser.selectedTab.hasAttribute('zen-empty-tab');
 }
 
+const ThemeCache = {
```
Move it on top of the function and call it `lazy`.

Not sure why you are doing your own implementation when one already exists, for example, here:

https://github.com/zen-browser/desktop/blob/96c6ec658cff4ec8b69192375ce8fbfc68737a29/src/zen/urlbar/ZenUBActionsProvider.sys.mjs#L32-L37

mr-cheffy commented on src/zen/urlbar/ZenUBGlobalActions.sys.mjs:
```diff
@@ -153,6 +173,30 @@ const globalActionsTemplate = [
+  },
+  {
+    label: 'Switch to Light Mode',
+    command: 'cmd_zenSwitchLightMode',
```
See my comment above, about using inline commands

nrysk committed:
fix: Remove boilerplate

nrysk commented:
Thank you for the detailed review! This was very helpful.
